app.py

```diff
@@ -88,68 +92,173 @@ def get_user_hours_form(ack, respond, body, command):
 @app.action("gethours_response")
 def get_logged_hours(ack, body, respond, logger):
     ack()
-    # Get list of users submitted for query by Slack admin
-    users = body['state']['values']['user_select_block']['user_select_input']['selected_users']
-    # Open an SQL connection
+    try:
+        users = body['state']['values']['user_select_block']['user_select_input']['selected_users']
+        start_date = body['state']['values']['date_select_block_start']['date_select_input']['selected_date']
+        end_date = body['state']['values']['date_select_block_end']['date_select_input']['selected_date']
+
+        if not (users and start_date and end_date):
+            raise ValueError("Missing required field")
+
+    except ValueError as e:
+        respond(f"*Invalid input, please try again!* {str(e)}.")
+        logger.warning(e)
+        return
+
     sqlc = database.SQLConnection()
     output = ""
     # Add the time logged by each user to the output
-    for i in users:
-        name = user_name(i)
-        user_time = sqlc.time_sum(i)
+    for user in users:
+        name = user_name(user)
+        user_time = sqlc.time_sum(user, start_date, end_date)
         if (user_time > 0):
-            output += f"*{name}*: {int(user_time/60)} hours and {int(user_time%60)} minutes\n"
+            output += f"*{name}*: {user_time//60} hours and {user_time%60} minutes\n"
         else:
             output += f"*{name}* has no logged hours\n"
-    # Send output to Slack chat and console
-    logger.info("\n" + output)
     respond(output)
 
+
+
 # Get user-selection form (choose users to see tables for their logged hours per date)
-@app.command("/getusertables")
+@app.command("/getentries")
 def get_user_hours_form(ack, respond, body, command):
     ack()
     if(is_admin(body['user_id'])):
-        respond(blocks=blocks.getusertables_form())
+        respond(blocks=blocks.getentries_form())
     else:
         respond("You must be an admin to use this command!")
 
 # Form response: log tables for all users specified
-@app.action("getusertables_response")
+@app.action("getentries_response")
 def get_logged_hours(ack, body, respond, logger):
     ack()
-    users = body['state']['values']['user_select_block']['user_select_input']['selected_users']
     try:
-        num_entries = re.findall(r'\d+', body['state']['values']['num_entries_block']['num_entries_input']['value'])[0]
-    except:
-        respond('Invalid input! Please try again.')
+        users = body['state']['values']['user_select_block']['user_select_input']['selected_users']
+        num_entries_input = body['state']['values']['num_entries_block']['num_entries_input']['value']
+
+        if not (users and num_entries_input):
+            raise ValueError("Missing required field")
+
+        try: num_entries = int(re.findall(r'\d+', num_entries_input)[0])
```

Comment:
I don't think we should be nesting try except blocks inside try blocks - is this inner one really necessary? If it is I think this should be refactored to avoid the nesting please